_posts/2016-11-29-future.md

```diff
@@ -0,0 +1,213 @@
+
+- [React plugins](https://babeljs.io/docs/plugins/#react)
+- [Flow plugin](https://babeljs.io/docs/plugins/transform-flow-strip-types)
+
+### Autoprefixer for Babel via [babel-preset-env](https://github.com/babel/babel-preset-env)
+
+This is something that is very important as we move forward to a point where users are compiling a specific set of transforms while the browsers support a specific set of features. There is a constant sliding window. This preset picks the right amount of plugins supported by the target environments so that the code can be fast/smaller.
+
+The goal of this effort is to make it simple to ship native code while allowing users to use the latest spec features. This will result in smaller payloads to users.
+
+The next big feature would be dynamic checking of polyfills as well as syntax as described in [babel/babel-preset-env#20](https://github.com/babel/babel-preset-env/issues/20).
+
+### Linting via [babel-eslint](https://github.com/babel/babel-eslint)
+
+As long as you want to lint javascript with experimental features (stage x, types) that ESLint proper won't support yet.
+
+Future: Better interop with ESLint (hook into scope, traversal), possibility of writing an ESLint rule with babel APIs, running ESLint rules as plugins/presets?
+
+### Minification
+
+[babel/babili](https://github.com/babel/babili) was created as a future proof minifier.
+
+It will be continually updated to take advantage of the latest javascript features (ES2015+ aware). Each transform can be independent and also pluggable.
+
+### Codemods/Refactoring
+
+Other relevant projects: [jscodeshift](https://github.com/facebook/jscodeshift), [recast](https://github.com/benjamn/recast), [Lebab](https://github.com/lebab/lebab).
+
+By passing in [recast](https://github.com/benjamn/recast) in the options, Babel can also be run on source code and can power the future of codemods. This feature was just made possible so we're looking forward to making it easier use and seeing the transformations it can enable.
+
+Check out the [6.16.0 blog post](https://babeljs.io/blog/2016/09/28/6.16.0#new-feature) for more info!
+
+### Code Coverage / Instrumentation
+
+With tools like [nyc](https://github.com/istanbuljs/nyc) and [babel-plugin-instanbul](https://github.com/istanbuljs/babel-plugin-istanbul).
+
+## Future
+
+### [Plugin API Changes](https://github.com/babel/notes/blob/master/2016-08/august-01.md#potential-api-changes-for-traversal)
+
+There is a lot of confusion around how plugins/presets interact regarding ordering. This results in bugs and issues with the config which require users to place plugins before/after others in a non-intuitive way.
+
+This is a fundamental change to the core of Babel which will take some work to figure out.
+
+### Typescript support
+
+Be able to understand typescript syntax (like we do with flow), add a plugin to strip typescript types for better interop.
```

Comment:
Might want to clarify whether or not this includes semantics of TypeScript. I imagine it does not, only syntax parsing, so I would guess a number of people will wrongfully assume Babel plans to literally add _real_ type checking support. The next section mentions using type information, but this is very very non-trivial to do for _everything_ (basically reimplementing tsc, as you prolly realize). TypeScript, unlike Flow, also requires runtime code for certain features like Enums, so it's not just stripping of types. Summary: "understand typescript syntax" might imply type checking to people who don't clearly understand the distinction.